addons/web_tour/static/src/tour_service/tour_utils.js

```diff
@@ -30,10 +30,13 @@ export function callWithUnloadCheck(func, ...args) {
  * @returns {HTMLElement | null}
  */
 export function getScrollParent(element) {
-    if (!element) {
+    if (!element || element.nodeType === 9) {
         return null;
     }
-    if (element.scrollHeight > element.clientHeight) {
+    const overflowY = window.getComputedStyle(element).overflowY;
+    const isScrollable = overflowY === "auto" || overflowY === "scroll" ||
+        (overflowY === "visible" && element === element.ownerDocument.scrollingElement);
+    if (isScrollable) {
```

Comment:
Probably deserve a comment: the previous condition seems right but probably is not because of the weird system we need for that snippet modal. If you don't want this reverted next week by the framework team, you better explain it right inside a beautiful comment :+1: